be/src/exec/parquet_writer.cpp

```diff
@@ -60,33 +67,397 @@ arrow::Status ParquetOutputStream::Tell(int64_t* position) const {
+                        return Status::InvalidArgument(ss.str());
+                    }
+                    LOG(WARNING) << "wangxixu-write-one-data:";
+                    parquet::RowGroupWriter* rgWriter = getRGWriter();
+                    parquet::BoolWriter *col_writer = static_cast<parquet::BoolWriter *>(rgWriter->column(index));
+                    if (item != nullptr) {
+                        col_writer->WriteBatch(1, nullptr, nullptr, static_cast<bool*> (item));
+                    } else {
+                        bool default_bool = false;
+                        col_writer->WriteBatch(1, nullptr, nullptr, &default_bool);
+                    }
+                    LOG(WARNING) << "wangxixu-write-finished0.";
+                    break;
+                }
+                case TYPE_TINYINT:
+                case TYPE_SMALLINT:
+                case TYPE_INT: {
+                    if (_str_schema[index][1] != "int32") {
+                        std::stringstream ss;
+                        ss << "project field type is tiny int/small int/int, should use int32, but the definition type of column "
+                                << _str_schema[index][2] << " is " << _str_schema[index][1];
+                        return Status::InvalidArgument(ss.str());
+                    }
+
+                    parquet::RowGroupWriter* rgWriter = getRGWriter();
+                    parquet::Int32Writer *col_writer = static_cast<parquet::Int32Writer *>(rgWriter->column(index));
+                    if (item != nullptr) {
+                        col_writer->WriteBatch(1, nullptr, nullptr, static_cast<int32_t *>(item));
+                    } else {
+                        int32_t default_int32 = 0;
+                        col_writer->WriteBatch(1, nullptr, nullptr, &default_int32);
+                    }
+                    break;
+                }
+                case TYPE_BIGINT: {
+                    if (_str_schema[index][1] != "int64") {
+                        std::stringstream ss;
+                        ss << "project field type is big int, should use int64, but the definition type of column "
+                            << _str_schema[index][2] << " is " << _str_schema[index][1];
+                        return Status::InvalidArgument(ss.str());
+                    }
+                    parquet::RowGroupWriter* rgWriter = getRGWriter();
+                    parquet::Int64Writer *col_writer = static_cast<parquet::Int64Writer *>(rgWriter->column(index));
+                    if (item != nullptr) {
+                        col_writer->WriteBatch(1, nullptr, nullptr, (int64_t * )(item));
+                    } else {
+                        int64_t default_int644 = 0;
+                        col_writer->WriteBatch(1, nullptr, nullptr, &default_int644);
+                    }
+                    break;
+                }
+                case TYPE_LARGEINT: {
```

Comment:
I think this should be checked on FE side, and reject request with LARGEINT column